### PR TITLE
fix: round robin seat booking rotation

### PR DESCRIPTION
**Pull request by anikdhabal**
# fix: round robin seat booking rotation

## Summary

Fixes the round robin seat booking bug where slots would disappear when all seats were taken for the current host, instead of rotating to the next available host in the round robin sequence. The system now properly creates separate booking records for different hosts when seats are full.

**Key Changes:**
- Modified `handleSeats.ts` to detect when a different host should handle the booking and skip finding existing bookings in those cases
- Updated `handleNewBooking.ts` to properly trigger round robin rotation when the current host's seats are full
- Enhanced `getUserAvailability.ts` to filter out existing seat bookings from busy times for round robin events
- Added comprehensive test coverage for the 2 hosts + 2 seats scenario

**Behavior Before:** Slots disappeared when first host's seats were full  
**Behavior After:** System rotates to next available host and creates separate booking records

## Review & Testing Checklist for Human

- [ ] **End-to-end test**: Create a seated round robin event with 2 hosts and 2 seats per slot. Verify that bookings 1&2 go to host 1, bookings 3&4 go to host 2, creating 2 separate booking records
- [ ] **Regression test**: Verify that existing seated events (non-round-robin) still work correctly and don't create separate bookings inappropriately  
- [ ] **Performance review**: Check that the additional database queries in the booking flow don't significantly impact booking performance
- [ ] **Code review**: Review the complex conditional logic in `handleSeats.ts` and `handleNewBooking.ts` for edge cases, especially around the `shouldCreateSeparateBooking` and `needsRoundRobinRotation` flags
- [ ] **Debug logging**: Decide whether the console.log statements should be removed before merging to production

**Recommended Test Plan:**
1. Set up a round robin team event with 2 hosts, 2 seats per slot
2. Book all 4 seats and verify 2 separate bookings are created with correct hosts
3. Test with existing seated events to ensure no regressions
4. Test edge cases like all hosts being unavailable or having booking conflicts

---

### Diagram

```mermaid
%%{ init : { "theme" : "default" }}%%
graph TB
    handleNewBooking["packages/features/bookings/lib/<br/>handleNewBooking.ts"]:::major-edit
    handleSeats["packages/features/bookings/lib/<br/>handleSeats/handleSeats.ts"]:::major-edit
    ensureAvailableUsers["packages/features/bookings/lib/<br/>handleNewBooking/ensureAvailableUsers.ts"]:::minor-edit
    getUserAvailability["packages/lib/<br/>getUserAvailability.ts"]:::minor-edit
    createNewSeat["packages/features/bookings/lib/<br/>handleSeats/create/createNewSeat.ts"]:::minor-edit
    findBookingQuery["packages/features/bookings/lib/<br/>handleNewBooking/findBookingQuery.ts"]:::minor-edit
    SeatsAvailabilityText["packages/features/bookings/components/<br/>SeatsAvailabilityText.tsx"]:::context
    testFile["packages/features/bookings/lib/handleNewBooking/<br/>test/team-bookings/seatedRoundRobin.test.ts"]:::minor-edit


    handleNewBooking -->|"calls when seats exist"| handleSeats
    handleNewBooking -->|"filters available users"| ensureAvailableUsers
    ensureAvailableUsers -->|"checks user availability"| getUserAvailability
    handleSeats -->|"creates new seat if same host"| createNewSeat
    handleSeats -->|"queries existing bookings"| findBookingQuery
    handleNewBooking -->|"displays seat counts"| SeatsAvailabilityText
    testFile -->|"tests"| handleNewBooking

    subgraph Legend
        L1["Major Edit"]:::major-edit
        L2["Minor Edit"]:::minor-edit  
        L3["Context/No Edit"]:::context
    end

    classDef major-edit fill:#90EE90
    classDef minor-edit fill:#87CEEB
    classDef context fill:#FFFFFF
```

### Notes

- The fix maintains backward compatibility with existing seated events (non-round-robin)
- Debug logging has been added to help troubleshoot any issues, but should be reviewed for production appropriateness
- The solution follows the existing round robin host selection patterns in the codebase
- Frontend seat counting already had protection against negative values via `Math.max(0, totalSeats - bookedSeats)`

**Session Details:**
- Requested by: anik@cal.com (@anikdhabal)
- Devin run: https://app.devin.ai/sessions/bd57d174a8f34f799e7d7923cf6dad93

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 1]
```diff
@@ -689,21 +689,39 @@ async function handler(
...
+          if (originalNonFixedHost) {
+            users = [...fixedHosts, originalNonFixedHost];
+          } else {
+            const attendeeEmailSet = new Set(booking.attendees.map((attendee) => attendee.email));
+            const nonFixedAttendeeHost = users.find(
+              (user) => !user.isFixed && attendeeEmailSet.has(user.email)
+            );
+            users = [...fixedHosts, ...(nonFixedAttendeeHost ? [nonFixedAttendeeHost] : [])];
+          }
+        }
```
Comment (anikdhabal) [inline packages/features/bookings/lib/handleNewBooking.ts]: DevinAI Isn't this the same as the code inside the else block on line 717? Why? Please fix it.

File: packages/features/bookings/lib/handleSeats/handleSeats.ts [hunk 2]
```diff
@@ -41,7 +41,48 @@ const handleSeats = async (newSeatedBookingObject: NewSeatedBookingObject) => {
 
   let resultBooking: HandleSeatsResultBooking = null;
 
-  const seatedBooking: SeatedBooking | null = await prisma.booking.findFirst({
+  let shouldCreateSeparateBooking = false;
+  
+  if (eventType.schedulingType === SchedulingType.ROUND_ROBIN && eventType.seatsPerTimeSlot && !rescheduleUid && !reqBookingUid) {
+    console.log(`[DEBUG] handleSeats - Checking for round robin separate booking logic. Organizer: ${evt.organizer.id}`);
```
Comment (anikdhabal) [inline packages/features/bookings/lib/handleSeats/handleSeats.ts]: DevinAI Remove all console logs from the changes in this PR.
